sims/verilator/Makefile

```diff
@@ -47,67 +47,133 @@ debug: $(sim_debug)
 include $(base_dir)/common.mk
 
 #########################################################################################
-# verilator binary and flags
+# verilator-specific user-interface variables and commands
 #########################################################################################
-VERILATOR := verilator --cc --exe
-
-CXXFLAGS := \
-	$(CXXFLAGS) -O1 -std=c++11 \
-	-I$(RISCV)/include \
-	-I$(dramsim_dir) \
-	-D__STDC_FORMAT_MACROS \
-	$(EXTRA_SIM_CC_FLAGS)
+HELP_COMPILATION_VARIABLES += \
+"   VERILATOR_PROFILE      = 'none' if no verilator profiling (default)" \
+"                            'all' if full verilator runtime profiling" \
+"                            'threads' if runtime thread profiling only" \
+"   VERILATOR_FST_MODE     = enable FST waveform instead of VCD. use with debug build"
 
-LDFLAGS := \
-	$(LDFLAGS) \
-	-L$(sim_dir) \
-	-lpthread
+#########################################################################################
+# verilator/cxx binary and flags
+#########################################################################################
+VERILATOR := verilator --cc --exe
 
-VERILATOR_CC_OPTS = \
+#----------------------------------------------------------------------------------------
+# user configs
+#----------------------------------------------------------------------------------------
+VERILATOR_PROFILE ?= none
+RUNTIME_PROFILING_CFLAGS := $(if $(filter $(VERILATOR_PROFILE),all),-g -pg,)
+RUNTIME_PROFILING_VFLAGS := $(if $(filter $(VERILATOR_PROFILE),all),\
+                              --prof-threads --prof-cfuncs,\
+                              $(if $(filter $(VERILATOR_PROFILE),threads),\
+								--prof-threads,))
+
+VERILATOR_FST_MODE ?= 0
+TRACING_OPTS := $(if $(filter $(VERILATOR_FST_MODE),0),\
+	                  --trace,--trace-fst --trace-threads 1)
+TRACING_CFLAGS := $(if $(filter $(VERILATOR_FST_MODE),0),,-DCY_FST_TRACE)
+
+#----------------------------------------------------------------------------------------
+# verilation configuration/optimization
+#----------------------------------------------------------------------------------------
+# we initially had --noassert for performance, but several modules use
+# assertions, including dramsim, so we enable --assert by default
+VMODE := \
 	-O3 \
-	-CFLAGS "$(CXXFLAGS) -DTEST_HARNESS=V$(VLOG_MODEL) -DVERILATOR" \
-	-CFLAGS "-I$(build_dir) -include $(build_dir)/$(long_name).plusArgs -include $(build_dir)/verilator.h" \
-	-LDFLAGS "$(LDFLAGS)" \
-	$(RISCV)/lib/libfesvr.a \
-	$(dramsim_lib)
-
-# default flags added for ariane
-ARIANE_VERILATOR_FLAGS = \
+	--x-assign fast \
+	--x-initial fast \
+	--assert \
+	--output-split 10000 \
+	--output-split-cfuncs 10000
+
+# default flags added for ariane (-Wno-fatal needed for -Wall to not cause
+# a crash, since 1000s of warnings are generated)
+VERILOG_IP_VERILATOR_FLAGS := \
 	--unroll-count 256 \
-	-Werror-PINMISSING \
```

Comment:
Whats the rational for removing these?

Reply:
These were removed by Sam in the past. I just copied this over.